js/GroupedBarChart.js

```diff
@@ -174,26 +195,161 @@ var GroupedBarChart = function(param)
                     .transition()
-                    .duration(150)
+                    .duration(barsAnimationTime/3)
                     // Turn back to original height
                     .attr('y', function(d) { return y(d.yval); })
-                    .attr('height', function(d) { return that.h - y(d.yval); })
+                    .attr('height', function(d) { return that.h - y(d.yval); });
+
+
+                    /*Data point funcitons*/
+                    var c_mouseover = function(d){
+                      d3.select(this)
+                          .classed('hover', true)
+                          .transition()
+                          .duration(400)
+                          .attr('r', 5 * 1.5)
+                          .transition()
+                          .duration(150)
+                          .attr('r', 5 * 1.25);
+                    };
+
+                    var c_mouseout = function(d){
+                      d3.select(this)
+                          .classed('hover', false)
+                          .transition()
+                          .duration(150)
+                          .attr('r', 5);
+                    };
+
+                    that.addLine = d3.svg.line()
+                        .x(
+                            function(d)
+                            {
+                                return x0(d.x0) + x1(d.x1) + x1.rangeBand()/2 ;
+                            })
+                        .y(
+                            function(d)
+                            {
+                                return y(d.y);
+                            });
+
+                    function createLineData(arrY, arrX0, arrX1){
+                      var lineData = [];
+                      var size = Math.max(arrY.length, arrX0.length, arrX1.length);
+                      for (var i = 0; i < size; i++) {
+                        var object= {
+                          "x0": arrX0[i],
+                          "x1": arrX1[0],
+                          "y": arrY[i]
+                        };
+                        if(_.values(object).length==3)
+                          lineData.push(object);
+                      }
+                      return lineData;
+                    }//end of functions
+
+                    /*data point data*/
+                    var circleAniminationTime = 500;
+                    var lineanimationTime = 1000;
+                    var lineData = createLineData(_.pluck(_.where(data,{name: "Total Average"}), "yval"),
+                    mainCategories, subCategories);
+                    console.log("lineData : " + JSON.stringify(lineData));
+
+
+                    /*Data Point render*/
+                    var gLine = that.svg.call(responsivefy)
+                        .append("g")
+                        .attr("class", "gline");
+
+                      /*
+                    var path = gLine.selectAll("path")
```

Comment:
Will remove the commented out code. Was trying different things Hingis to try to get the line to be a rainbow pattern from left to right